lib/errors.ts

```diff
@@ -20,7 +20,9 @@ import * as _ from 'lodash';
 import * as os from 'os';
 import * as Raven from 'raven';
 
-import * as patterns from './utils/patterns';
+export class ExpectedError extends Error {}
```

Comment:
This commit (changes to `errors.ts` and `patterns.ts`) introduces the `ExpectedError` type. An "expected" error is an error we explicitly test for (e.g. some data validation), and which does not require Sentry.io reporting. We currently have a "situation" in the CLI where:
* Several commands are arguably incorrectly handling "expected" errors with `throw Error("some reason")`.
* Most commands follows the arguably correct pattern of calling the `exitWithExpectedError()` function (which calls `process.exit(1)` before anything is reported to Sentry.io).

The former is deemed incorrect, and the latter correct, simply because the top-level `handleError()` function in `errors.ts` would report any error that reached it to Sentry.io. However, [ordinarily calling process.exit() is not a recommended practice](https://stackoverflow.com/a/37592669) as it may prematurely abort async operations (like perhaps updating config files - theoretically it could cause data corruption). When possible, it is preferable to set `process.exitCode` and allow the process to exit gracefully. This commit modifies the `handleError()` function so that it does not report instances of `ExpectedError` to Sentry.io, and set `process.exitCode` instead of calling `process.exit()` for expected errors. This allows `throw ExpectedError("some reason")` as an alternative correct way of handling errors in CLI commands.